src/Raven.Client/Documents/Subscriptions/AbstractSubscriptionWorker.cs

```diff
@@ -603,29 +556,77 @@ private async Task ProcessSubscriptionInternalAsync(JsonContextPool contextPool,
                         {
                             if (_logger.IsInfoEnabled)
                             {
-                                _logger.Info(
-                                    $"Subscription '{_options.SubscriptionName}'. Subscriber threw an exception on document batch", ex);
+                                _logger.Info($"Subscription '{_options.SubscriptionName}'. Subscriber threw an exception on document batch", ex);
                             }
 
-                            if (_options.IgnoreSubscriberErrors == false)
-                                throw new SubscriberErrorException($"Subscriber threw an exception in subscription '{_options.SubscriptionName}'", ex);
+                            HandleSubscriberError(ex);
                         }
                     }
 
-                    try
+                    await SendAckAsync(batch, tcpStreamCopy, context, _processingCts.Token).ConfigureAwait(false);
+                });
+            }
+        }
+
+        protected virtual void HandleSubscriberError(Exception ex)
+        {
+            if (_options.IgnoreSubscriberErrors == false)
+                throw new SubscriberErrorException($"Subscriber threw an exception in subscription '{_options.SubscriptionName}'", ex);
+        }
+
+        internal virtual async Task<BatchFromServer> PrepareBatchAsync(JsonContextPool contextPool, Stream tcpStreamCopy, JsonOperationContext.MemoryBuffer buffer, TBatch batch, Task notifiedSubscriber)
```

Comment:
protected?

Reply:
no we use `BatchFromServer` which is internal